ArtOfIllusion/src/artofillusion/MaterialMappingDialog.java

```diff
@@ -39,28 +39,30 @@ public MaterialMappingDialog(BFrame parent, Object3D obj)
   {
     super(parent, "Material Mapping", true);
 
-    fr = parent;
     this.obj = obj;
     map = obj.getMaterialMapping();
     oldMapping = map.duplicate();
 
     // Make a list of all material mappings which can be used for this object and material.
 
-    mappings = new Vector();
-    List<MaterialMapping> allMappings = PluginRegistry.getPlugins(MaterialMapping.class);
-    for (int i = 0; i < allMappings.size(); i++)
+    mappings = new Vector<Class>();
```

Comment:
@peastman Here is another reason to move away from reflection if we can. The reflection call on line 51 will not compile if we try to Parameterise this vector to anything more specific than Class...